faunadb/client_test.go

```diff
@@ -1802,6 +1903,73 @@ func (s *ClientTestSuite) TestEvalTruncExpression() {
 	s.Require().Equal(1.23, num)
 }
 
+func (s *ClientTestSuite) TestEvalAnyAllExpressions() {
+	var b []bool
+
+	s.queryAndDecode(f.Arr{
+		f.Any(f.Arr{true, true, false}),
+		f.All(f.Arr{true, true, true}),
+		f.Any(f.Arr{false, false, false}),
+		f.All(f.Arr{true, true, false}),
+	}, &b)
+
+	s.Require().Equal([]bool{true, true, false, false}, b)
+}
+
+func (s *ClientTestSuite) TestEvalCountMeanSumExpression() {
+	var expected f.Arr
+	data := f.Arr{1, 2, 3, 4, 5, 6, 7, 8, 9}
+
+	col := s.queryForRef(f.CreateCollection(f.Obj{"name": "countmeansum_test"}))
+	s.queryForRef(f.CreateIndex(f.Obj{"name": "countmeansum_idx", "source": col, "active": true, "values": f.Arr{f.Obj{"field": f.Arr{"data", "value"}}}}))
+	s.query(
+		f.Foreach(data, f.Lambda("x", f.Create(col, f.Obj{"data": f.Obj{"value": f.Pow(f.Var("x"), 2)}}))),
+	)
+
+	m := f.Match(f.Index("countmeansum_idx"))
+
+	s.queryAndDecode(f.Arr{
+		f.Count(data),
+		f.Mean(data),
+		f.Sum(data),
+	}, &expected)
+
+	s.Require().Equal(f.Arr{f.LongV(9), f.DoubleV(5), f.LongV(45)}, expected)
+
+	s.queryAndDecode(f.Arr{
+		f.Count(m),
+		f.Trunc(f.Mean(m)),
+		f.Sum(m),
+	}, &expected)
+
+	s.Require().Equal(f.Arr{f.LongV(9), f.DoubleV(31.66), f.DoubleV(285)}, expected)
+
+}
+
+func (s *ClientTestSuite) TestEvalDocumentsExpression() {
+	var i int
+
+	aCollection := f.RandomStartingWith("collection_")
+	anIndex := f.RandomStartingWith("index_")
+
+	s.query(f.CreateCollection(f.Obj{"name": aCollection}))
+	s.query(f.CreateIndex(f.Obj{"name": anIndex, "source": f.Collection(aCollection), "active": true}))
+
+	counter := 0
+	maxCount := 27
+	data := make([]int, maxCount)
+
+	for counter < maxCount {
+		data[counter] = counter + 1
+		counter++
+	}
+
+	s.query(f.Foreach(data, f.Lambda("x", f.Create(f.Collection(aCollection), f.Obj{"data": f.Obj{"count": f.Var("x")}}))))
```

Comment:
you if are just `Count`ing the documents, you don't need to add `data`, just an empty object is okay